arangod/VocBase/Methods/Collections.cpp

```diff
@@ -1289,19 +1289,31 @@ arangodb::Result Collections::checksum(LogicalCollection& collection,
   return trx.finish(res);
 }
 
+/// @brief the list of collection attributes that are allowed by user-input
+/// this is to avoid retyping the same list twice
+#define COMMON_ALLOWED_COLLECTION_INPUT_ATTRIBUTES                            \
+  StaticStrings::DataSourceSystem, StaticStrings::DataSourceId, "keyOptions", \
+      StaticStrings::WaitForSyncString, StaticStrings::CacheEnabled,          \
+      StaticStrings::ShardKeys, StaticStrings::NumberOfShards,                \
+      StaticStrings::DistributeShardsLike, "avoidServers",                    \
+      StaticStrings::IsSmart, StaticStrings::ShardingStrategy,                \
+      StaticStrings::GraphSmartGraphAttribute, StaticStrings::Schema,         \
+      StaticStrings::SmartJoinAttribute, StaticStrings::ReplicationFactor,    \
+      StaticStrings::MinReplicationFactor, /* deprecated */                   \
+      StaticStrings::WriteConcern, "servers"
+
 arangodb::velocypack::Builder Collections::filterInput(
-    arangodb::velocypack::Slice properties) {
+    arangodb::velocypack::Slice properties, bool allowDC2DCAttributes) {
+#ifdef USE_ENTERPRISE
+  if (allowDC2DCAttributes) {
```

Comment:
Probably, we could add a TRI_ASSERT for cluster check here?

Reply:
I think for now yes, DC-2-DC only replicates clusters.
However I think there is no technical reason for this piece to only work in Cluster environment, so I would not like to include the ASSERT here, it is required in other places on DC2DC, where it matters that we are on a cluster.